server/controllers/app.controller.spec.js

```diff
@@ -184,6 +188,8 @@ describe('The app controller', function() {
 
     this.sandbox.stub(modulesMocks.redisClient, 'storeShopAsync');
     this.sandbox.stub(modulesMocks.redisClient, 'quitAsync');
+    this.sandbox.stub(modulesMocks.redisClient, 'getShopAsync')
+    .returns(Promise.resolve({ accessToken: '127424ab9aa0ebce26dfdc786bc7fba4'}));
```

Comment:
I think that we should take advantage of this test to verify that we are not paying attention to the _accessToken_ from the _session_.

What do you think about this:

```diff
  it('connectToDoppler should call put Doppler integration using Doppler Client', async function() {
    const shop = 'store.myshopify.com';
-   const accessToken = '127424ab9aa0ebce26dfdc786bc7fba4';
+   const storedAccessToken = '127424ab9aa0ebce26dfdc786bc7fba4';
+   const sessionAccessToken = 'WRONG123456';
    const request = sinonMock.mockReq({
      session: { shop, accessToken: sessionAccessToken },
      body: {
        dopplerApiKey: 'C22CADA13759DB9BBDF93B9D87C14D5A',
        dopplerAccountName: 'user@example.com',
      },
    });
    const response = sinonMock.mockRes();

    this.sandbox.stub(modulesMocks.redisClient, 'storeShopAsync');
    this.sandbox.stub(modulesMocks.redisClient, 'quitAsync');
    this.sandbox.stub(modulesMocks.redisClient, 'getShopAsync')
    .returns(Promise.resolve({ accessToken: storedAccessToken}));
    // . . .
    expect(modulesMocks.dopplerClient.putShopifyIntegrationAsync)
      .to.have.been.callCount(1);
    expect(modulesMocks.dopplerClient.putShopifyIntegrationAsync)
-     .to.be.called.calledWithMatch(shop, accessToken);
+     .to.be.called.calledWithMatch(shop, sessionAccessToken);
    expect(response.sendStatus)
      .to.be.called.calledWithExactly(200);
```

Reply:
Looks good